examples/optimize/optimize.py

```diff
@@ -262,19 +345,14 @@ def evaluate(cfg: DictConfig):
     agent = make_agent(cfg)
     env = make_env()
     dataset = get_dataset(cfg)
-    tapes_save_path = f"dev_tapes_{cfg.dataset.dev_size}.yaml"
-    final_tapes = batch_run_and_save(agent, env, dataset, tapes_save_path)
-    retrieval_accuracy = compute_retrieval_accuracy(dataset.dev, final_tapes)
-    answer_accuracy = compute_answer_exact_match(dataset.dev, final_tapes)
-    print(f"Retrieval accuracy: {retrieval_accuracy:.2f}")
-    print(f"Answer accuracy: {answer_accuracy:.2f}")
-    metrics_save_path = f"metrics_{cfg.dataset.dev_size}.json"
-    with open(metrics_save_path, "w") as f:
-        json.dump({"retrieval_accuracy": retrieval_accuracy, "answer_accuracy": answer_accuracy}, f, indent=2)
+    tapes_save_path = f"test_tapes_{cfg.dataset.test_size}.yaml"
+    final_tapes = batch_run_and_save(agent, env, dataset.test, tapes_save_path)
+    metric_mean_retrieval_answer(dataset.test, final_tapes, run_name="test")
```

Comment:
maybe rename this function to `evaluate_and_save_metrics`?

Reply:
I wanted to get a function name that explains what metric is being calculated as we pass these metric evaluation function to the optimize_demo function. I save the metric as part of being observable but there might be a better pattern for systematic metric logging we could use later on.